core/images.go

```diff
@@ -189,3 +190,69 @@ func (n *OpenBazaarNode) GetBase64Image(url string) (base64ImageData, filename s
 	_, filename = path.Split(u.Path)
 	return img, filename, nil
 }
+
+func (n *OpenBazaarNode) maybeMigrateImageHashes(listing *pb.Listing) error {
+	if listing.Item == nil || len(listing.Item.Images) == 0 {
+		return nil
+	}
+
+	for i, image := range listing.Item.Images {
+		largeID, err := cid.Decode(image.Large)
+		if err != nil {
+			return err
+		}
+		if largeID.Version() > 0 {
+			hash, err := ipfs.AddFile(n.IpfsNode, path.Join(n.RepoPath, "root", "images", "large", image.Filename))
+			if err != nil {
+				return err
+			}
+			image.Large = hash
+		}
+		mediumID, err := cid.Decode(image.Medium)
+		if err != nil {
+			return err
+		}
+		if mediumID.Version() > 0 {
+			hash, err := ipfs.AddFile(n.IpfsNode, path.Join(n.RepoPath, "root", "images", "medium", image.Filename))
+			if err != nil {
+				return err
+			}
+			image.Medium = hash
+		}
+		smallID, err := cid.Decode(image.Small)
+		if err != nil {
+			return err
+		}
+		if smallID.Version() > 0 {
+			hash, err := ipfs.AddFile(n.IpfsNode, path.Join(n.RepoPath, "root", "images", "small", image.Filename))
+			if err != nil {
+				return err
+			}
+			image.Small = hash
+		}
+		tinyID, err := cid.Decode(image.Tiny)
+		if err != nil {
+			return err
+		}
+		if tinyID.Version() > 0 {
+			hash, err := ipfs.AddFile(n.IpfsNode, path.Join(n.RepoPath, "root", "images", "tiny", image.Filename))
+			if err != nil {
+				return err
+			}
+			image.Tiny = hash
+		}
+		originalID, err := cid.Decode(image.Original)
+		if err != nil {
+			return err
+		}
+		if originalID.Version() > 0 {
+			hash, err := ipfs.AddFile(n.IpfsNode, path.Join(n.RepoPath, "root", "images", "original", image.Filename))
+			if err != nil {
+				return err
+			}
+			image.Original = hash
+		}
+		listing.Item.Images[i] = image
+	}
+	return nil
```

Comment:
Change this behavior so it's not repeating every time. Extract this behavior into a helper or a local anon function:
```
originalID, err := cid.Decode(image.Original)
		if err != nil {
			return err
		}
		if originalID.Version() > 0 {
			hash, err := ipfs.AddFile(n.IpfsNode, path.Join(n.RepoPath, "root", "images", "original", image.Filename))
			if err != nil {
				return err
			}
			image.Original = hash
		}
```